### DISCUSSION TITLE
[Feature Request]: Allow passing additional data to cloud init to create files and run scripts

### DISCUSSION THREAD

**Discussion by rombert** [mysticaltech/terraform-hcloud-kube-hetzner#1039] [Ideas]
### Description

This was discussed at https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/discussions/948 . Briefly, my goal is to be able to configure and start a custom Wireguard service on all provisioned nodes. To do that, I think we should expose two additional variables and end up in 

- write_files: https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/e79cfc775fc1fb6752b31c299487aa568f48390a/modules/host/templates/cloudinit.yaml.tpl#L5
- runcmd: https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/e79cfc775fc1fb6752b31c299487aa568f48390a/modules/host/templates/cloudinit.yaml.tpl#L25

The variables should initially be empty and have per-host values, since Wireguard keys cannot be shared.

I plan to work on a pull request sometime in October, but I won't mind if someone else works on it first :-)

**Comment by mysticaltech** [comment]
@M4t7e @ifeulner @aleksasiriski Please, I would like your input on that people 🙏

**Comment by mysticaltech** [comment]
@rombert Is your current implementation open-source? If so, please do share your modified cloud-init.yaml.tpl, if not, no worries.

**Comment by rombert** [comment]
@mysticaltech - I don't have a modification yet, I just tested this out my manually speciying cloud-init data for one VM. Nothing done in terraform yet.

**Comment by mysticaltech** [comment]
(Quick tip, I do not know how familiar you are with terraform, but GPT-4 sure is a decent expert, I personally use it all the time now 😄)

**Comment by Silvest89** [comment]
@mysticaltech It should be configurable per node and not nodepool? This is fairly easy to implement though

**Reply by mysticaltech** [reply]
@Silvest89  It would make sense to me per nodepool indeed! PR most welcome 🙏

**Comment by mysticaltech** [comment]
Implemented in PR: https://github.com/mysticaltech/terraform-hcloud-kube-hetzner/pull/2070

Closing this discussion as addressed for the v3 workstream.
